Library/Homebrew/cmd/audit.rb

```diff
@@ -622,22 +623,31 @@ def audit_specs
     end
   end
 
-  def audit_revision
+  def audit_revision_and_version_scheme
     return unless formula.tap # skip formula not from core or any taps
     return unless formula.tap.git? # git log is required
 
     fv = FormulaVersions.new(formula, :max_depth => 10)
-    revision_map = fv.revision_map("origin/master")
-    revisions = revision_map[formula.version]
-    if !revisions.empty?
-      problem "revision should not decrease" if formula.revision < revisions.max
-    elsif formula.revision != 0
-      if formula.stable
-        if revision_map[formula.stable.version].empty? # check stable spec
-          problem "revision should be removed"
+    attributes_map = {}
+    attributes_map[:revision] = fv.revision_map("origin/master")
+    attributes_map[:version_scheme] = fv.version_scheme_map("origin/master")
+
+    attributes_map.keys.each do |attribute|
+      attribute_map = attributes_map[attribute]
+      attributes_for_version = attribute_map[formula.version]
+      formula_attribute_value = formula.send(attribute)
+      if !attributes_for_version.empty?
+        if formula_attribute_value < attributes_for_version.max
+          problem "#{attribute} should not decrease"
+        end
+      elsif formula_attribute_value != 0
+        if formula.stable # check stable spec
+          if attributes_for_version[formula.stable.version].empty?
+            problem "#{attribute} should be removed"
+          end
+        else # head/devel-only formula
+          problem "#{attribute} should be removed"
         end
-      else # head/devel-only formula
-        problem "revision should be removed"
```

Comment:
This is incorrect. `version_scheme` is different than `revision` as it should never be removed or decreased.